### PR TITLE
Common handlers

### PR DESCRIPTION
**Pull request by einarmo**
Code to generalize the CleanFromError and CleanRequest methods. These contain a lot of common logic in most cases, so we can create a common base method and call that, reducing complexity and code duplication.

This could be made easier with native SDK methods for getting Identity from the DTOs.

### CODE REVIEW

**Comment by cognite-cicd** [comment]
#### Checkmarx scan completed with the following findings
|Lines|Severity|Category|File|Link|
---|---|---|---|---
[68](https://github.com/cognitedata/dotnet-extractor-utils/blob/common-handlers/Cognite.Metrics/MetricsConfig.cs#L68) |Medium|Heap_Inspection|Cognite.Metrics/MetricsConfig.cs|[Checkmarx](https://cognite.checkmarx.net/CxWebClient/ViewerMain.aspx?scanid=1038884&projectid=413&pathid=1)

File: Cognite.Extensions/Sanitation.cs [hunk 1]
```diff
@@ -233,5 +233,104 @@ private static int SafeByteCount(string? str)
...
+
+        private static (List<T>, List<CogniteError<T>>) CleanRequest<T>(
+            DistinctResource<T>[] distinctResources,
+            IEnumerable<T> items,
+            Func<T, ResourceType?> verify,
+            Action<T> sanitize,
+            SanitationMode mode)
+        {
+            if (mode == SanitationMode.None) return (items.ToList(), new List<CogniteError<T>>());
+            if (items == null) throw new ArgumentNullException(nameof(items));
```
Comment (silvavelosa) [inline Cognite.Extensions/Sanitation.cs:260]:
```suggestion
            if (items == null) throw new ArgumentNullException(nameof(items));
            if (mode == SanitationMode.None) return (items.ToList(), new List<CogniteError<T>>());
```

**Review by silvavelosa — APPROVED** [review]
LGTM ✔️

File: Cognite.Extensions/Sanitation.cs (hunk 1 above)
Comment (einarmo) [reply Cognite.Extensions/Sanitation.cs:260]: Nice catch

**Pull request merged**